subsys/bluetooth/host/adv.c

```diff
@@ -1332,6 +1357,20 @@ int bt_le_ext_adv_delete(struct bt_le_ext_adv *adv)
 #endif /* defined(CONFIG_BT_EXT_ADV) */
 
 
+static void adv_timeout(struct k_work *work)
+{
+#if defined(CONFIG_BT_EXT_ADV)
+	struct bt_le_ext_adv *adv = CONTAINER_OF(work, struct bt_le_ext_adv,
+	timeout_work);
```

Comment:
This almost works; `work` is a `struct k_work`, but `timeout_work` is a `struct k_work_delayable`. 

You can get the `struct k_work_delayable` either by doing another `CONTAINER_OF` manually, or use 
`k_work_delayable_from_work` (which is just a wrapper for a `CONTAINER_OF`).